protocol/logger/job_logger.go

```diff
@@ -59,28 +60,22 @@ func (logger *WebsocketLivelogger) Connect() error {
 	if logger.Connection.Trace {
 		fmt.Printf("Try to connect to websocket %s\n", logger.FeedStreamUrl)
 	}
-	origin, err := url.Parse(logger.Connection.TenantURL)
-	if err != nil {
-		return err
-	}
 	re := regexp.MustCompile("(?i)^http(s?)://")
 	feedStreamUrl, err := url.Parse(re.ReplaceAllString(logger.FeedStreamUrl, "ws$1://"))
 	if err != nil {
 		return err
 	}
-	logger.ws, err = websocket.DialConfig(&websocket.Config{
-		Location: feedStreamUrl,
-		Origin:   origin,
-		Version:  13,
-		Header: http.Header{
+	logger.ws, _, err = websocket.Dial(context.Background(), feedStreamUrl.String(), &websocket.DialOptions{
+		HTTPHeader: http.Header{
 			"Authorization": []string{"Bearer " + logger.Connection.Token},
+			"User-Agent": []string{"github-act-runner/1.0.0"},
 		},
 	})
 	return err
 }
 
 func (logger *WebsocketLivelogger) SendLog(lines *protocol.TimelineRecordFeedLinesWrapper) error {
-	return websocket.JSON.Send(logger.ws, lines)
+	return wsjson.Write(context.Background(), logger.ws, lines)
```

Comment:
Thank you for guiding me here as I was not sure how you use contexts. I did add 5min timeout, but I believe it would be better to scope this from some main context which might be little bit challenging?